macOS/debugger.cpp

```diff
@@ -505,6 +509,24 @@ void *Debugger::RemoteAllocateNear(uint64_t region_min,
   return ret_address;
 }
 
+void *Debugger::RemoteAllocate(size_t size, MemoryProtection protection) {
+  mach_vm_address_t alloc_address = 0;
+  vm_prot_t protection_flags = MacOSProtectionFlags(protection);
+
+  kern_return_t krt = mach_vm_allocate(mach_target->Task(),
+                                       (mach_vm_address_t*)&alloc_address,
+                                       size,
+                                       VM_FLAGS_ANYWHERE);
+
+  if (krt != KERN_SUCCESS) {
+    FATAL("Unable to allocate memory region starting @ %p, size 0x%lx\n",
```

Comment:
NIT: The message here shouldn't include alloc_address, as in this case we are not attempting to allocate on a specific address.

Reply:
good catch!